codegen/models/project_foundation/implementor_simple_alg_info.xml

```diff
@@ -0,0 +1,41 @@
+<implementor name="alg info" is_default="1">
+    Provide implementation of algorithm information for algorithm agility
+    <implementation name="simple alg info">
+        Provide simple algorithm information (hash) implementation
+        <interface name="alg info"/>
+
+        <context>
+            <property name="alg id" enum="alg id"/>
+        </context>
+
+        <constructor name="set alg id" visibility="public">
+            Set algorithm identificator
+            <argument name="alg id" enum="alg id"/>
+        </constructor>
+
+    </implementation>
+
+    <implementation name="kdf alg info">
+        Provide KDF algorithm information implementation
+        <interface name="alg info"/>
+
+        <context>
+            <property name="alg id" enum="alg id"/>
+            <property name="hash alg info" impl="simple alg info"/>
+            <property name="salt" class="buffer"/>
+            <property name="iteration count" type="size"/>
+        </context>
+
+        <constructor name="set alg id" visibility="public">
+            Set algorithm identificator
+            <argument name="alg id" enum="alg id"/>
+        </constructor>
+
+        <constructor name="set alg id and hash id" visibility="public">
+            Set algorithm identificator
+            <argument name="alg id" enum="alg id"/>
+            <argument name="hash alg info" impl="simple alg info" access="readwrite"/>
```

Comment:
Use `access="retain"` or `access="disown"` because this object is stored for later use.
Use `access="readwrite"` only if some object state can be changed and an object itself is not stored for future use.